src/ORM/Locator/LocatorAwareTrait.php

```diff
@@ -70,7 +70,7 @@ public function setTableLocator(LocatorInterface $tableLocator)
      */
     public function getTableLocator()
     {
-        if ($this->_tableLocator === null) {
```

Comment:
How can it throw undefined property error when the trait itself has `$_tableLocator` declared?
https://github.com/cakephp/cakephp/blob/4c543ecd4a8e2d6e2db7fdf65ef3bebef3722175/src/ORM/Locator/LocatorAwareTrait.php#L30

Reply:
Declared does not mean that it is defined.

Reply:
http://sandbox.onlinephpfunctions.com/code/dfc84393b4ecf03d9d3bf2dd1a3cd0a3d5cca838

Reply:
Are you trying to troll me with that pastebin?  You are using local var `$_b` instead of property `$this->_b`.

Reply:
This would be a valid example which doesn't throw any error http://sandbox.onlinephpfunctions.com/code/dfc84393b4ecf03d9d3bf2dd1a3cd0a3d5cca838

```
<?php
class A {
    
    protected $_b;
    
    public function getB() {
        if ($this->_b === null) {
            $this->_b = true;
        }
        return $this->_b;
    }
    
}

$a = new A();
$a->getB();
```

Reply:
My bad in the sandbox. I wonder why do I get undefined in LocatorAwareTrait.php:73.

Reply:
No problem :slightly_smiling_face:  Do leave a comment here when you figure out what's causing the issue for you. We have similar property === null checks at multiple places throughout the core.

Reply:
It seems that `unset($this->_tableLocator);` was called at some point in our code.